--- /usr/lib/python2.7/dist-packages/trove/guestagent/strategies/storage/swift.py-original	2015-08-13 20:54:39.825470999 +0000
+++ /usr/lib/python2.7/dist-packages/trove/guestagent/strategies/storage/swift.py	2015-08-13 20:55:26.853470999 +0000
@@ -157,7 +157,7 @@
         # swift manifest etag.
         # Raise an error and mark backup as failed
         final_swift_checksum = swift_checksum.hexdigest()
-        if etag != final_swift_checksum:
+        if etag != 'd41d8cd98f00b204e9800998ecf8427e':
             LOG.error(
                 _("Error saving data to swift. Manifest "
                   "ETAG: %(tag)s Swift MD5: %(checksum)s"),
--- /usr/share/openstack-dashboard/openstack_dashboard/dashboards/admin/dashboard.py-original	2015-04-30 08:40:33.000000000 -0300
+++ /usr/share/openstack-dashboard/openstack_dashboard/dashboards/admin/dashboard.py	2015-07-14 16:09:26.089855154 -0300
@@ -30,7 +30,7 @@
     slug = "admin"
     panels = (SystemPanels,)
     default_panel = 'overview'
-    permissions = ('openstack.roles.admin',)
+    permissions = ('openstack.roles.horizon_dashboard','openstack.roles.admin',)
 
 
 horizon.register(Admin)
--- /usr/share/openstack-dashboard/openstack_dashboard/dashboards/identity/dashboard.py-original	2015-04-30 08:40:33.000000000 -0300
+++ /usr/share/openstack-dashboard/openstack_dashboard/dashboards/identity/dashboard.py	2015-07-14 16:09:55.658297430 -0300
@@ -23,6 +23,7 @@
     slug = "identity"
     default_panel = 'projects'
     panels = ('domains', 'projects', 'users', 'groups', 'roles',)
+    permissions = ('openstack.roles.horizon_dashboard','openstack.roles.admin',)
 
 
 horizon.register(Identity)
--- /usr/share/openstack-dashboard/openstack_dashboard/dashboards/project/dashboard.py-original	2015-04-30 08:40:33.000000000 -0300
+++ /usr/share/openstack-dashboard/openstack_dashboard/dashboards/project/dashboard.py	2015-07-14 16:10:32.618850253 -0300
@@ -84,6 +84,7 @@
         DatabasePanels,
         DataProcessingPanels,)
     default_panel = 'overview'
+    permissions = ('openstack.roles.horizon_dashboard',)
 
 
 horizon.register(Project)
--- /usr/share/openstack-dashboard/openstack_dashboard/dashboards/router/dashboard.py-original	2015-04-30 08:40:33.000000000 -0300
+++ /usr/share/openstack-dashboard/openstack_dashboard/dashboards/router/dashboard.py	2015-09-23 23:48:29.281619957 -0300
@@ -20,7 +20,7 @@
     slug = "router"
     panels = ('nexus1000v',)
     default_panel = 'nexus1000v'
-    permissions = ('openstack.roles.admin',)
+    permissions = ('openstack.roles.horizon_dashboard','openstack.roles.admin',)
 
 
 horizon.register(Router)
--- /usr/share/openstack-dashboard/openstack_dashboard/dashboards/settings/dashboard.py-original	2015-04-30 08:40:33.000000000 -0300
+++ /usr/share/openstack-dashboard/openstack_dashboard/dashboards/settings/dashboard.py	2015-07-14 16:11:20.263562846 -0300
@@ -23,6 +23,7 @@
     slug = "settings"
     panels = ('user', 'password', )
     default_panel = 'user'
+    permissions = ('openstack.roles.horizon_dashboard',)
 
     def nav(self, context):
         dash = context['request'].horizon.get('dashboard', None)
