--- /usr/share/openstack-dashboard/openstack_dashboard/dashboards/admin/dashboard.py-original	2015-02-05 15:06:50.000000000 -0200
+++ /usr/share/openstack-dashboard/openstack_dashboard/dashboards/admin/dashboard.py	2015-05-07 23:32:50.447398393 -0300
@@ -30,7 +30,7 @@
     slug = "admin"
     panels = (SystemPanels,)
     default_panel = 'overview'
-    permissions = ('openstack.roles.admin',)
+    permissions = ('openstack.roles.horizon_dashboard','openstack.roles.admin',)
 
 
 horizon.register(Admin)
--- /usr/share/openstack-dashboard/openstack_dashboard/dashboards/identity/dashboard.py-original	2015-02-05 15:06:44.000000000 -0200
+++ /usr/share/openstack-dashboard/openstack_dashboard/dashboards/identity/dashboard.py	2015-05-07 23:33:42.944700531 -0300
@@ -23,6 +23,7 @@
     slug = "identity"
     default_panel = 'projects'
     panels = ('domains', 'projects', 'users', 'groups', 'roles',)
+    permissions = ('openstack.roles.horizon_dashboard','openstack.roles.admin',)
 
 
 horizon.register(Identity)
--- /usr/share/openstack-dashboard/openstack_dashboard/dashboards/project/dashboard.py-original	2015-05-07 23:41:32.668352342 -0300
+++ /usr/share/openstack-dashboard/openstack_dashboard/dashboards/project/dashboard.py	2015-05-07 23:31:25.649295108 -0300
@@ -83,6 +83,7 @@
         DataProcessingPanels,)
     default_panel = 'overview'
     supports_tenants = True
+    permissions = ('openstack.roles.horizon_dashboard',)
 
 
 horizon.register(Project)
--- /usr/share/openstack-dashboard/openstack_dashboard/dashboards/router/dashboard.py-original	2015-05-07 23:42:46.666188037 -0300
+++ /usr/share/openstack-dashboard/openstack_dashboard/dashboards/router/dashboard.py	2015-05-07 23:31:25.673295702 -0300
@@ -22,7 +22,7 @@
     slug = "router"
     panels = ('nexus1000v',)
     default_panel = 'nexus1000v'
-    permissions = ('openstack.roles.admin',)
+    permissions = ('openstack.roles.horizon_dashboard','openstack.roles.admin',)
 
 
 horizon.register(Router)
--- /usr/share/openstack-dashboard/openstack_dashboard/dashboards/settings/dashboard.py-original	2015-05-07 23:43:30.615278314 -0300
+++ /usr/share/openstack-dashboard/openstack_dashboard/dashboards/settings/dashboard.py	2015-05-07 23:36:03.420184969 -0300
@@ -23,6 +23,7 @@
     slug = "settings"
     panels = ('user', 'password', )
     default_panel = 'user'
+    permissions = ('openstack.roles.horizon_dashboard',)
 
     def nav(self, context):
         dash = context['request'].horizon.get('dashboard', None)
